components/Pagination/Pagination.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+const MAX_VISIBLE_LINKS = 7;
+
+const Separator = () => (
+  <span className='padding-small text--bold pagination-default'>...</span>
+);
+
+
+class Pagination extends Component {
+  constructor() {
+    super();
+    this.link = this.link.bind(this);
+    this.links = this.links.bind(this);
+  }
+
+  link(i) {
+    const { currentIndex, onChange } = this.props;
+    return (
+      <span className={`pagination-button text--bold padding-vert-xsmall padding-horz-small radius ${currentIndex === i ? 'active bg-gray-5 text--white' : ''}`} onClick={() => onChange(i)} key={`link-${i}`}>
+        {i + 1}
+      </span>
+    );
+  }
+
+  links() {
+    const { currentIndex, length } = this.props;
+    const { link } = this;
+    const truncateBefore = currentIndex > 3 && length > MAX_VISIBLE_LINKS;
+    const truncateAfter = currentIndex < length - 4 && length > MAX_VISIBLE_LINKS;
+    const links = [];
+    if (truncateBefore) links.push(<Separator key='sep-0' />);
+    if (truncateBefore && truncateAfter) {
```

Comment:
Hrm, on second look, this might not actually help that much. I was thinking something to the effect of links = arr.slice(currentIndex, 3), but since you'd have to generate the links anyway, this would just replace separate pushes with a loop.